pkg/http/http.go

```diff
@@ -61,15 +62,8 @@ func (r *CustomRoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 	if resp != nil {
```

Comment:
Just a memo. When ported this method, not added any test coverage

TODO: in follow up

<img width="1740" height="550" alt="Screenshot 2025-08-01 at 08 07 32" src="https://github.com/user-attachments/assets/27c482f3-8af7-43ce-bac4-85aed644019a" />

Reply:
I've gone and created a PR that adds coverage for this method and a couple others that were missing it #5712